### ISSUE TITLE
[Question/Issue] Regarding execution status

### ISSUE THREAD

**Issue by Alan-pad** [phsmith/rundeck_exporter#11]
Hello,

I've just begun using your exporter and I was wondering why am i getting duplicated metrics regarding execution status, is this intended ? I though I would get only the last execution status of a job but I'm getting multiple and don't really understand why.

- Example :

**Version** : 2.2.3

**Environment variables** :

```yaml
- name: RUNDECK_EXPORTER_HOST
  value: "0.0.0.0"
- name: RUNDECK_EXPORTER_PORT
  value: "9620"
- name: RUNDECK_PROJECTS_EXECUTIONS
  value: "True"
- name: RUNDECK_PROJECTS_EXECUTIONS_CACHE
  value: "True"
- name: RUNDECK_SKIP_SSL
  value: "True"
- name: RUNDECK_URL
  value: "http://127.0.0.1:4440"
```

**Results of** `curl http://127.0.0.1:9620`:

> I'm having those metrics returned ~ 15 times

```
# HELP rundeck_project_execution_status Rundeck Project Main Execution Status
# TYPE rundeck_project_execution_status gauge
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="failed"} 0.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="unknown"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="failed"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="unknown"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="succeeded"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="failed"} 1.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="unknown"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="failed"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="unknown"} 0.0
# HELP rundeck_project_execution_status Rundeck Project Main Execution Status
# TYPE rundeck_project_execution_status gauge
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="failed"} 0.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="1c653977-acfb-4c4d-9571-2d63367afcc9",job_name="job-1",project_name="Main",status="unknown"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="failed"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="7bdd9002-f498-4ea1-9ee7-42d640649f5b",job_name="job-2",project_name="Main",status="unknown"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="succeeded"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="failed"} 1.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="dd935e23-9f41-436b-bbe6-6e45c1332fb0",job_name="job-3",project_name="Main",status="unknown"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="succeeded"} 1.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="running"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="failed"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="aborted"} 0.0
rundeck_project_execution_status{job_id="ed5245f9-a06f-40db-953d-77d747d0ffe4",job_name="job-4",project_name="Main",status="unknown"} 0.0
```

**Comment by Alan-pad** [comment]
While I'm waiting for your answer, I'll try to find the source of this behaviour.

**Comment by phsmith** [comment]
Hi @Alan-pad,

I'm working right now to fix that wrong behavior.

**Comment by Alan-pad** [comment]
Oh alright, let me know if you need help I have some time this afternoon (Paris time)

**Comment by phsmith** [comment]
@Alan-pad,

I've just published a new version, 2.2.4, with the fix for the problem with duplicated metrics. 

When you have a time, make a validation please.

**Comment by Alan-pad** [comment]
@phsmith Thanks a lot it's fixed :pray:
